src/utils/convertMarkdown.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { Fragment } from "react";
+
+import ReactMarkdown from "react-markdown";
+import glm from "remark-gfm";
+import emoji from "remark-emoji";
+import externalLinks from "remark-external-links";
+import sanitize from "rehype-sanitize";
+
+export const convertMarkdown = (
+  text: string | undefined,
+  options?: { images?: boolean; allowP?: boolean }
```

Comment:
```suggestion
  options?: { allowImages?: boolean; allowP?: boolean }
```

Reply:
As per https://github.com/sparkletown/sparkle/pull/1383#issuecomment-844729468, we probably don't need to have `allowP`. I expect it's fine for it to be hardcoded to always use `Fragment`.

Reply:
wouldn't that collapse separate paragraphs together eg. for some (not in chat) text, e.g. description for the poster or whatnot?
Do we have any markdown multi-paragraph example uploaded anywhere within the platform DB to check it out @margulies ?

Reply:
> wouldn't that collapse separate paragraphs together eg. for some (not in chat) text, e.g. description for the poster or whatnot?

Yeah.. you're probably right about that actually..

---

Taking a step back from this, can you remind me what the issue with showing the wrapping `p` was? Was it that it broke the visual look of the UI? Or just that the generated code looked uglier?

If it's just the code, then that's probably not something we need to be concerned about here.

Reply:
> Do we have any markdown multi-paragraph example uploaded anywhere within the platform DB to check it out

I think it should be fairly straightforward to create a multiline event/etc description or similar to test this out.

Reply:
> Taking a step back from this, can you remind me what the issue with showing the wrapping p was?

In the chat it resulted in an extra blank line at the end of each chat message

Reply:
Cool, thanks. I wonder if we can just tweak the styling a bit to factor for that (eg. it's probably some `margin` or `padding` or similar I expect) and not have to remove the `p` entirely from here? If so, then that might be a better solution (and likely would allow for multiline messages to work)

Reply:
As long as chat doesn't allow a new line to be entered (as it is now) I don't see how allowing for p and figuring out how to counteract p with styling would be useful.

Reply:
Because this same component that renders markdown is intended to be used across the platform in fields where we absolutely want multiline support (as that was one of the original use cases that lead to this exploration to begin with).

And while the chat may not support multiline at the moment, making it able to do so would take trivial code changes (swapping `input` element for `textarea` or similar + maybe some local keybindings so that `shift+enter` inserts a newline rather than sending the message, etc; similar to how it works in most chat apps), so if it's just a tweak to the styles to correct the 'wrapped in `p`' issue here, then the not-overengineering-the-solution approach would be to just make that localised tweak fix, rather than changing the fundamentals of the react renderer and the elements it uses across the entire application.

Reply:
But there is allowP and everywhere else markdown is used without saying it to False. What am I still missing? But actually, may be over we could just deal with it in css and override all needed styles... I will check although know no css much
I would welcome shift+enter support in chat, but didn't dare to suggest ;-)

Reply:
> I would welcome shift+enter support in chat, but didn't dare to suggest ;-)

https://github.com/Andarist/react-textarea-autosize should work to give us a  a textarea that can expand. Min rows should be set to 1 so it stays small by default, and then set some sane upper limit so it doesn’t get too tall even when there’s a lot of text (at which point I expect it should scroll)

To add the shift+enter keyboard shortcut we’ll ideally want to fix useMousetrap so that it can properly use a ref. I’ll try and look into that when I’m on a computer next and see if I can figure the solution for it.

Reply:
re allowP: pushed 120f343d234730c503a19b77c4ad3f90a569bcf8 where I removed allowP and tuned css to my liking.  The only gotcha - could not overload styling for pre, code.  But I think it would be ok.  

<details>
<summary>Here is how some Hs and overall messages look like now, where "another" is H1</summary> 

![image](https://user-images.githubusercontent.com/39889/119517368-8cf21300-bd45-11eb-83dc-c0f96a0e6d36.png)


</details>

Reply:
re Shift-Enter + autosize ... not sure if I would be able to get to it (but may be @soichih or @margulies would find time to help), but hopefully we don't need to make this PR conditioned on that extra feature.

Reply:
Definitely can be OOS of this PR, but would be good to capture the relevant notes/details from here in a backlog issue that we can reference/build off in future

Reply:
> edit: it would be also not as useful unless #1429 lands

That PR seems unrelated?

Reply:
d'oh -- I think I meant #1421 PR closing #1392 issue -- resizeable chat sidebar. That is when having multiple lines in a single post formatted nicely with images etc would be handy.  ATM it would just clutter than chat message etc IMHO.

Reply:
> Shift enter auto size definitely can be OOS of this PR, but would be good to capture the relevant notes/details from here in a backlog issue that we can reference/build off in future

Created https://github.com/sparkletown/sparkle/issues/1451 to track this.

Reply:
> ATM it would just clutter than chat message etc IMHO.

Mm, makes sense. I've added this as a note on #1451 as well.